javagen/src/main/java/com/azure/autorest/mapper/ProxyMethodMapper.java

```diff
@@ -177,8 +182,50 @@ protected void buildUnexpectedResponseExceptionTypes(ProxyMethod.Builder builder
                                                          Operation operation, List<HttpResponseStatus> expectedStatusCodes,
                                                          JavaSettings settings) {
         ClassType errorType = null;
+        Map<ClassType, List<HttpResponseStatus>> errorTypeMap = new HashMap<>();
+
+        /*
+        1. If exception has valid numeric status codes, group them to unexpectedResponseExceptionTypes
+        2. If exception does not have status codes, or have 'default' or invalid number, put the first to unexpectedResponseExceptionType, ignore the rest
+        3. After processing, if no model in unexpectedResponseExceptionType, take any from unexpectedResponseExceptionTypes and put it to unexpectedResponseExceptionType
+         */
+
         if (operation.getExceptions() != null && !operation.getExceptions().isEmpty()) {
-            errorType = (ClassType) Mappers.getSchemaMapper().map(operation.getExceptions().get(0).getSchema());
+            //errorType = (ClassType) Mappers.getSchemaMapper().map(operation.getExceptions().get(0).getSchema());
+            for (Response exception : operation.getExceptions()) {
+                boolean isDefaultError = true;
+                if (exception.getProtocol() != null && exception.getProtocol().getHttp() != null) {
+                    List<String> statusCodes = exception.getProtocol().getHttp().getStatusCodes();
+                    if (statusCodes != null && !statusCodes.isEmpty() && exception.getSchema() != null) {
+                        try {
+                            ClassType errorTypeWithStatusCodes = (ClassType) Mappers.getSchemaMapper().map(exception.getSchema());
+                            List<HttpResponseStatus> statusCodeList = statusCodes.stream()
+                                    .map(code -> HttpResponseStatus.valueOf(Integer.parseInt(code)))
+                                    .collect(Collectors.toList());
+
+                            if (errorTypeMap.containsKey(errorTypeWithStatusCodes)) {
+                                errorTypeMap.get(errorTypeWithStatusCodes).addAll(statusCodeList);
+                            } else {
+                                errorTypeMap.put(errorTypeWithStatusCodes, statusCodeList);
+                            }
+                        } catch (NumberFormatException ex) {
+                            // statusCodes can be 'default'
```

Comment:
If statusCode is `default`, then shouldn't `isDefaultError` be true (assuming there are no other integer statusCodes)? Currently, it would be marked as false.

Reply:
Yes, you are correct. Fixing.